core/state_transition.go

```diff
@@ -278,6 +285,9 @@ func (st *stateTransition) buyGas() error {
 		}
 	}
 	balanceCheck.Add(balanceCheck, st.msg.Value)
+	if operatorCost != nil {
+		balanceCheck.Add(balanceCheck, operatorCost.ToBig())
+	}
```

Comment:
In the case that `gasFeeCap == nil` then `balanceCheck = gasLimit * gasPrice + l1Cost + operatorCost + value + operatorCost`, note that operator cost is added twice. Therefore I believe it would be correct to move the lines:

```
	if operatorCost != nil {
		balanceCheck.Add(balanceCheck, operatorCost.ToBig())
	}
```

Within the conditional `if st.msg.GasFeeCap != nil {`.

Reply:
Done